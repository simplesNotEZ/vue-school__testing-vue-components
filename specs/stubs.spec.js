import { shallowMount } from '@vue/test-utils';
import ListComponent from '@/list';

const ListItemStub = {
  template: `<li>{{ movie }}</li>`,
  props: ['movie']
};

describe('ListComponent', () => {
  test('shallowMount', () => {
    const wrapper = shallowMount(ListComponent, {
      stubs: {
        ListItem: ListItemStub
      }
    });
    // expect(wrapper.html()).toMatchSnapshot();
    // Remember, you don't have to use the .html() method off of wrapper
    // since we have imported and make use of (via jest.config.js in
    // snapshotSerializers: []) the jest-serializer-vue package
    expect(wrapper).toMatchSnapshot();
  });
});
