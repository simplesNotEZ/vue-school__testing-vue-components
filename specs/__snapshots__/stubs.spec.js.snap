// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListComponent shallowMount 1`] = `
<div>
  <ul>
    <li>Iron Man</li>
    <li>Avengers</li>
    <li>Infinity War</li>
  </ul>
</div>
`;
